src/main/java/net/pms/encoders/YoutubeDl.java

```diff
@@ -34,6 +36,31 @@
 
 public class YoutubeDl extends FFMpegVideo {
 
+	String extractVideoId(String url) {
```

Comment:
It would be good to rename this to indicate it is YouTube-specific, because youtube-dl supports a lot of other websites too (a future goal could be to add support for all of the sites)